be/src/http/utils.cpp

```diff
@@ -125,17 +139,15 @@ void do_file_response(const std::string& file_path, HttpRequest* req) {
         HttpChannel::send_error(req, HttpStatus::NOT_FOUND);
         return;
     }
-    struct stat st;
-    auto res = fstat(fd, &st);
-    if (res < 0) {
+
+    int64_t file_size = get_file_size(fd);
```

Comment:
Have you considered using the `Env` APIs?

Reply:
Thanks for the tip, it has been revised.